cmd/libs/go2idl/go-to-protobuf/protobuf/generator.go

```diff
@@ -526,7 +560,7 @@ func membersToFields(locator ProtobufLocator, t *types.Type, localPackage types.
 			if len(field.Name) == 0 && len(parts[0]) != 0 {
 				field.Name = parts[0]
 			}
-			if field.Name == "-" {
+			if field.Tag == -1 && field.Name == "-" {
```

Comment:
A field that has name `-` and a tag assigned is supposed to be in the
protobuf IDL, but not in JSON.  We have a few objects where we don't use
the JSON serialization because it's impossible, so we mark the field as
ignored in protobuf unless someone explicitly set a tag (previously).

On Fri, Jan 22, 2016 at 3:11 PM, Paul Morie <notifications@github.com>
wrote:

> In cmd/libs/go2idl/go-to-protobuf/protobuf/generator.go
> <https://github.com/kubernetes/kubernetes/pull/19426#discussion_r50584783>
> :
>
> > @@ -526,7 +560,7 @@ func membersToFields(locator ProtobufLocator, t *types.Type, localPackage types.
> >  			if len(field.Name) == 0 && len(parts[0]) != 0 {
> >  				field.Name = parts[0]
> >  			}
> > -			if field.Name == "-" {
> > +			if field.Tag == -1 && field.Name == "-" {
>
> Why is the additional check needed on tag?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/19426/files#r50584783>.
>